src/Engine/Waypoint/Waypoints.cpp

```diff
@@ -104,16 +104,22 @@ void
 Waypoints::WaypointNameTree::Add(WaypointPtr wp)
 {
   TCHAR normalized_name[wp->name.length() + 1];
+  TCHAR normalized_shortname[wp->shortname.length() +1];
```

Comment:
This is a variable-length array which is illegal in C++.
The line above it does the same, but it's also bad. We should fix all VLAs in our code, and not write new code using this feature.

Reply:
I replaced this vla with a dynamic array. I left the original code, as this cleanup isn't part of this feature.